allowtree.go

```diff
@@ -0,0 +1,105 @@
+package firewall
+
+import (
+	"bytes"
+	"fmt"
+	"net"
+)
+
+// AllowIPTree is a data structure for storing ip addresses and searching them efficiently
+type AllowIPTree struct {
```

Comment:
should we make this Allow/Block list the same type, and just call it `IPFilter` type and specifyc `Allow: bool` on it, which would represent this as either allow or block rule..? the reason is, someone might want to pass a range for the allow list too..? or "1.2.3.4/32" .. what are pros/cons of keeping the AllowIPTree as it is now?